### PR TITLE
Add Miroof's Virtual Gamepad

### PR DESCRIPTION
**Pull request by HerbFargus**
script that installs a virtual gamepad from Miroof (https://github.com/miroof/node-virtual-gamepads) with node.js that enables people to use their smartphone as a gamepad. The code is probably not the most standardised but it was the only way I was able to figure out how to get it to work. if you have any suggestions for improvement I'm all ears.

### CODE REVIEW

**Comment by HerbFargus** [comment]
Ok, thanks for helping me figure it out. I've ammended it and tested it on a fresh build and it is working great!

**Comment by joolswills** [comment]
please can you have a go at squashing these commits and pushing back to this pull request? :)

**Comment by HerbFargus** [comment]
Ha ha so now you want me to squash them... Ha ha sure. I'll give it a go.

**Comment by joolswills** [comment]
well, in this case it's more logical to do so - as it's a restructure in the second commit that isn't needed in the history etc.

**Comment by joolswills** [comment]
basically, on your branch

```
git rebase -i 5d8dd2a8da7c230486396b3e311f20b6f6719122 (the commit before your first commit)
```
edit the second line and change it from pick to squash - save the file.

and you then have a single commit. and you can push -f back 

BTW you could do one more commit first and then squash, as you have some trailing whitespace in the file

**Comment by joolswills** [comment]
You also have incorrect indentation in the install function.

File: scriptmodules/supplementary/virtualgamepad.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+rp_module_id="virtualgamepad"
+rp_module_desc="Virtual Gamepad for Smartphone"
+rp_module_menus="4+"
+
+function sources_virtualgamepad() {
+    wget http://node-arm.herokuapp.com/node_latest_armhf.deb
+    dpkg -i node_latest_armhf.deb
+    rm node_latest_armhf.deb
+}
```
Comment (joolswills) [inline scriptmodules/supplementary/virtualgamepad.sh]: I would probably put this in the install section too - as it isn't source but a binary package.

**Comment by HerbFargus** [comment]
When you are referring to the trailing whitespaces- were you referring to EOL? I'm editing things in notepad++ on a windows PC- should I change the EOL format to unix instead of windows?

**Comment by joolswills** [comment]
I believe git is handling EOL conversion automatically, as the files end up as unix line endings at my end. I was referring to the extra spaces you had after `cd "$md_inst"` before the end of the line.

**Comment by HerbFargus** [comment]
ok. that should do it I think. Thanks for being patient with me

**Comment by joolswills** [comment]
Thanks!

**Pull request merged**